packages/discord.js/src/structures/Attachment.js

```diff
@@ -3,9 +3,16 @@
 const Util = require('../util/Util');
 
 /**
- * Represents an attachment.
+ * Represents an attachment
  */
 class Attachment {
+  /**
+   * @typedef {Object} AttachmentPayload
+   * @property {?string} name
+   * @property {Stream|BufferResolvable} attachment
+   * @property {?string} description
+   */
+
   /**
    * @param {BufferResolvable|Stream} attachment The file
```

Comment:
The constructor could be changed to only accept an `APIAttachment`, as all three constructor calls are doing `new Attachment(a.url, a.filename, a)`.

Since it's not a builder anymore, the constructor could be private too.